app/models/saleable_day.rb

```diff
@@ -11,9 +11,10 @@ class SaleableDay < ApplicationRecord
   validates :date, presence: true
 
   class << self
+    # gives priority to child who put date up for sale / buy first
+    # this could be a starting point for working on priority refinement
     def find_first_available_on_date(saleable_day_id)
-      days = SaleableDay.where(buyer_id: nil)
-      saleable_day = days.find(saleable_day_id)
+      saleable_day = SaleableDay.find(saleable_day_id)
```

Comment:
will we lose something by losing this check?

Reply:
That is a good question. So, I think this might be an artifact that I should have refactor out but did not. I'm not entirely sure. I went through a couple of iterations around how to do this with `SaleableDays`. When I released the first iteration of the app and got feedback from people I realized that all of the requirements had not been conveyed to me. So, once it was in people's hands I was giving a number of additional things that need to be done/supported. Hence, I made a couple of significant changes, if I remember correctly. 

This `Saleable.where(buyer_id: nil)` doesn't seem to make a whole lot of sense. So, no. I don't think anything is lost by not having it.